src/main/java/dev/dirs/impl/Windows.java

```diff
@@ -0,0 +1,199 @@
+package dev.dirs.impl;
+
+import dev.dirs.Constants;
+
+import java.lang.foreign.AddressLayout;
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.GroupLayout;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SymbolLookup;
+import java.lang.foreign.ValueLayout;
+import java.lang.invoke.MethodHandle;
+
+import static dev.dirs.impl.Util.isNullOrEmpty;
+import static dev.dirs.impl.Util.stringLength;
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_CHAR;
+
+public final class Windows {
+
+	private Windows() {}
+
+	static {
+		if (Constants.operatingSystem == 'w') {
+			System.loadLibrary("ole32");
+			System.loadLibrary("shell32");
+		}
+	}
+
+	private static final SymbolLookup SYMBOL_LOOKUP = SymbolLookup.loaderLookup().or(Linker.nativeLinker().defaultLookup());
+	private static final ValueLayout.OfByte C_CHAR = ValueLayout.JAVA_BYTE;
+	private static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT;
+	private static final AddressLayout C_POINTER = ValueLayout.ADDRESS
+			.withTargetLayout(MemoryLayout.sequenceLayout(java.lang.Long.MAX_VALUE, JAVA_BYTE));
+	private static final ValueLayout.OfInt C_LONG = ValueLayout.JAVA_INT;
+
+	public static String getProfileDir() {
+		return getDir("{5E6C858F-0E22-4760-9AFE-EA3317B67173}");
+	}
+
+	public static String getMusicDir() {
+		return Windows.getDir("{4BD8D571-6D19-48D3-BE97-422220080E43}");
+	}
+
+	public static String getDesktopDir() {
+		return Windows.getDir("{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}");
+	}
+
+	public static String getDocumentsDir() {
+		return Windows.getDir("{FDD39AD0-238F-46AF-ADB4-6C85480369C7}");
+	}
+
+	public static String getDownloadsDir() {
+		return Windows.getDir("{374DE290-123F-4565-9164-39C4925E467B}");
+	}
+
+	public static String getPicturesDir() {
+		return Windows.getDir("{33E28130-4E1E-4676-835A-98395C3BC3BB}");
+	}
+
+	public static String getPublicDir() {
+		return Windows.getDir("{DFDF76A2-C82A-4D63-906A-5644AC457385}");
+	}
+
+	public static String getTemplatesDir() {
+		return Windows.getDir("{A63293E8-664E-48DB-A079-DF759E0509F7}");
+	}
+
+	public static String getVideosDir() {
+		return Windows.getDir("{18989B1D-99B5-455B-841C-AB7C74E4DDFC}");
+	}
+
+	public static String getRoamingAppDataDir() {
+		return getDir("{3EB685DB-65F9-4CF6-A03A-E3EF65729F3D}");
+	}
+
+	public static String getLocalAppDataDir() {
+		return getDir("{F1B32785-6FBA-4FCF-9D55-7B8E7F157091}");
+	}
+
+	public static String getDir(String folderId) {
+		try (var arena = Arena.ofConfined()) {
+			MemorySegment guidSegment = arena.allocate(GUID_LAYOUT);
+			if (CLSIDFromString(createSegmentFromString(folderId, arena), guidSegment) != 0) {
+				System.exit(-1);
```

Comment:
`return null` or throwing exception instead?

Reply:
Yeah - definitely don't want to exit. That was just for my little utility program.